src/ray/core_worker/future_resolver.h

```diff
@@ -30,9 +32,11 @@ namespace ray {
 class FutureResolver {
  public:
   FutureResolver(std::shared_ptr<CoreWorkerMemoryStore> store,
+                 std::shared_ptr<LocalityDataReporterInterface> locality_data_reporter,
```

Comment:
Could we use a lambda instead? I think interfaces are generally better once there are multiple methods that should be logically grouped together.